ui/app/components/transit-key-actions.js

```diff
@@ -35,213 +50,142 @@ const TRANSIT_PARAMS = {
-  export: 'Exported your key',
-};
-export default Component.extend(TRANSIT_PARAMS, {
-  store: service(),
-  flashMessages: service(),
-
-  // public attrs
-  selectedAction: null,
-  key: null,
-  isModalActive: false,
-
-  onRefresh() {},
-  init() {
-    this._super(...arguments);
-    // TODO figure out why get is needed here Ember Upgrade
-    // eslint-disable-next-line ember/no-get
-    if (this.selectedAction) {
-      return;
-    }
-    // eslint-disable-next-line ember/no-get
-    set(this, 'selectedAction', get(this, 'key.supportedActions.firstObject'));
-    assert('`key` is required for `' + this.toString() + '`.', this.getModelInfo());
-  },
-
-  didReceiveAttrs() {
-    this._super(...arguments);
-    this.checkAction();
-    if (this.selectedAction === 'export') {
-      this.setExportKeyDefaults();
-    }
-  },
-
-  setExportKeyDefaults() {
-    const exportKeyType = this.key.exportKeyTypes.firstObject;
-    const exportKeyVersion = this.key.validKeyVersions.lastObject;
-    this.setProperties({
-      exportKeyType,
-      exportKeyVersion,
-    });
-  },
-
-  keyIsRSA: computed('key.type', function () {
-    const type = this.key.type;
-    return type === 'rsa-2048' || type === 'rsa-3072' || type === 'rsa-4096';
-  }),
 
-  getModelInfo() {
-    const model = this.key || this.backend;
-    if (!model) {
-      return null;
-    }
-    const backend = model.backend || model.id;
-    const id = model.id;
-
-    return {
-      backend,
-      id,
-    };
-  },
-
-  checkAction() {
-    const currentAction = this.selectedAction;
-    const oldAction = this.oldSelectedAction;
-
-    this.resetParams(oldAction, currentAction);
-    set(this, 'oldSelectedAction', currentAction);
-  },
-
-  resetParams(oldAction, action) {
-    const params = copy(TRANSIT_PARAMS);
-    let paramsToKeep;
-    const clearWithoutCheck =
-      !oldAction ||
-      // don't save values from datakey
-      oldAction === 'datakey' ||
-      // can rewrap signatures — using that as a ciphertext later would be problematic
-      (oldAction === 'rewrap' && !this.key.supportsEncryption);
-
-    if (!clearWithoutCheck && action) {
-      paramsToKeep = PARAMS_FOR_ACTION[action];
-    }
-
-    if (paramsToKeep) {
-      paramsToKeep.forEach((param) => delete params[param]);
-    }
-    //resets params still left in the object to defaults
-    this.clearErrors();
-    this.setProperties(params);
-    if (action === 'export') {
-      this.setExportKeyDefaults();
-    }
-  },
-
-  handleError(e) {
-    this.set('errors', e.errors);
-  },
+export default class TransitKeyActions extends Component {
+  @service store;
+  @service flashMessages;
+  @service router;
 
-  clearErrors() {
-    this.set('errors', null);
-  },
+  @tracked isModalActive = false;
+  @tracked errors = null;
+  @tracked props = { ...STARTING_TRANSIT_PROPS }; // Shallow copy of the object. We don't want to mutate the original.
+  @tracked submitIsRunning = this.doSubmit.isRunning;
```

Comment:
Does this work how expected? It seems like this would just be set to false on init and not update when the task initiates. Can we directly access `this.doSubmit.isRunning` in the template instead?

Reply:
Ah, good catch. I'll amend.